integration/src/integration/java/pl/allegro/tech/hermes/integration/ConsumingHttp2Test.java

```diff
@@ -15,56 +15,92 @@
 
 import javax.net.ssl.SSLContext;
 import javax.ws.rs.core.Response;
-
-import java.security.NoSuchAlgorithmException;
 import java.util.concurrent.atomic.AtomicInteger;
 
 import static io.undertow.UndertowOptions.ENABLE_HTTP2;
 import static javax.ws.rs.core.Response.Status.CREATED;
+import static org.assertj.core.api.Assertions.fail;
 import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
 
 public class ConsumingHttp2Test extends IntegrationTest {
-    static final int HTTPS_PORT = Ports.nextAvailable();
-    static final String HTTPS_ENDPOINT = "https://localhost:" + HTTPS_PORT;
+
+    private final static String MESSAGE_BODY = TestMessage.of("hello", "h2").body();
 
     private AtomicInteger incomingCounter = new AtomicInteger(0);
-    private Undertow http2Server;
 
     @Test
-    public void shouldDeliverMessageUsingHttp2() throws InterruptedException {
+    public void shouldDeliverMessageUsingHttp2() {
         // given
-        Topic topic = operations.buildTopic("deliverHttp2", "topic");
+        int httpsPort = Ports.nextAvailable();
+        String httpsEndpoint = "https://localhost:" + httpsPort;
 
-        Subscription subscription = SubscriptionBuilder.subscription(topic, "subscription")
-                .withEndpoint(HTTPS_ENDPOINT)
-                .withHttp2Enabled(true)
-                .build();
+        Undertow http2Server = http2Server(httpsPort,
+                exchange -> exchange.getResponseSender().send(Integer.toString(incomingCounter.incrementAndGet())));
 
-        operations.createSubscription(topic, subscription);
+        http2Server.start();
+
+        Topic topic = createTopicAndSubscription(httpsEndpoint);
 
         // when
-        Response response = publisher.publish(topic.getQualifiedName(), TestMessage.of("hello", "h2").body());
+        Response response = publisher.publish(topic.getQualifiedName(), MESSAGE_BODY);
 
         // then
         assertThat(response).hasStatus(CREATED);
         wait.until(() -> assertThat(incomingCounter.intValue()).isPositive());
     }
 
+    @Test
+    public void shouldDeliverMessageWithoutKeepAliveHeaderUsingHttp2() {
+        // given
+        int httpsPort = Ports.nextAvailable();
+        HttpString keepAliveHeader = new HttpString("Keep-Alive");
+        String httpsEndpoint = "https://localhost:" + httpsPort;
+
+        Undertow http2Server = http2Server(httpsPort, exchange -> {
+            if (exchange.getRequestHeaders().contains(keepAliveHeader)) {
```

Comment:
maybe it would be worth checking the protocol, so we'll be absolutely sure it's http2. it should be available in the exchange.

Reply:
Okay